### PR TITLE
Test add custom errors on helpers

### PR DESCRIPTION
**Pull request by eloycoto**
Added a new helper error message, that has a callback function that can be called when the Ginkgo trigger an error, so the errors dump to users are more powerful.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4859)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: test/helpers/errors.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// callback function
+func NewSSHMetaError(s string, callback func() string) *SSHMetaError {
+	return &SSHMetaError{
+		s:       s,
+		errorCB: callback,
+	}
+}
+
+// SSHMetaError is a custom error that executes a callback function each time
+// that it's called Error() function
```
Comment (ianvernon) [inline test/helpers/errors.go]: Documentation is not a valid phrase; "executes a callback function when its Error() function is invoked" ?

File: test/helpers/errors.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+type SSHMetaError struct {
+	s       string
```
Comment (ianvernon) [inline test/helpers/errors.go]: Field name is not descriptive. why not change it to "errorString"?

File: test/helpers/errors.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	errorCB func() string
```
Comment (ianvernon) [inline test/helpers/errors.go]: I don't like this abbreviation. Call it "errorCallBack".

File: test/helpers/errors.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package helpers
+
+import "fmt"
+
+// NewSSHMetaError returns a new SSHMetaError with the given string and a
+// callback function
+func NewSSHMetaError(s string, callback func() string) *SSHMetaError {
+	return &SSHMetaError{
+		s:       s,
+		errorCB: callback,
```
Comment (ianvernon) [inline test/helpers/errors.go]: Check if the provided callback is nil; If it is nil, put a dummy-callback in place to ensure that no panic occurs when `String()` is called, which unconditionally invokes the function. Or, you can ensure that errorCB is not nil in functions which utilize it.

File: test/helpers/kubectl.go [hunk 5]
```diff
@@ -878,9 +877,28 @@ func (kub *Kubectl) CiliumPolicyAction(namespace, filepath string, action Resour
 	err = WithTimeout(
 		body,
 		"cannot change state of resource correctly; command timed out",
-		&TimeoutConfig{Timeout: timeout})
+		&TimeoutConfig{Timeout: 1})
```
Comment (ianvernon) [inline test/helpers/kubectl.go]: What is the need for this change? This is unrelated to the content in the commit message, apart from that it utilizes the new SSHMetaError. Factor it out into a separate commit and explain why the change is necessary in the commit message.

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: test/helpers/kubectl.go (hunk 5 above)
Comment (eloycoto) [reply test/helpers/kubectl.go]: ouch, embarrassing, my bad!

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]
